examples/opensora_hpcai/opensora/models/vae/vae.py

```diff
@@ -49,3 +52,302 @@ def encode_with_moments_output(self, x):
+        # This is a mistake made during the training of OpenSora v1.2.
+        # To re-use the trained model, we need to keep this mistake.
+        # For training, we should refine to 0.13025.
+        self.scale_factor = 0.18215
+
+    @staticmethod
+    def rearrange_in(x):
+        B, C, T, H, W = x.shape
+        # (b c t h w) -> (b t c h w)
+        x = ops.transpose(x, (0, 2, 1, 3, 4))
+        x = ops.reshape(x, (B * T, C, H, W))
+
+        return x
+
+    @staticmethod
+    def rearrange_out(x, B):
+        # x = rearrange(x, "(B T) C H W -> B C T H W", B=B)
+        BT, C, H, W = x.shape
+        T = BT // B
+        x = ops.reshape(x, (B, T, C, H, W))
+        x = ops.transpose(x, (0, 2, 1, 3, 4))
+
+        return x
+
+    def encode(self, x):
+        # x: (B, C, T, H, W)
+        # NOTE: remind to use stop gradient when invoke it
+        # is_video = (x.ndim == 5)
+
+        B = x.shape[0]
+        # x = rearrange(x, "B C T H W -> (B T) C H W")
+        x = self.rearrange_in(x)
+
+        if self.micro_batch_size is None:
+            # x = self.module.encode(x).latent_dist.sample().mul_(0.18215)
+            x = self.module.encode(x) * self.scale_factor
+        else:
+            bs = self.micro_batch_size
+            x_out = []
+            # FIXME: supported in graph mode? or use split
+            for i in range(0, x.shape[0], bs):
+                x_bs = x[i : i + bs]
+                x_bs = self.module.encode(x_bs) * self.scale_factor
+                x_out.append(x_bs)
+            x = ops.cat(x_out, axis=0)
+
+        # x = rearrange(x, "(B T) C H W -> B C T H W", B=B)
+        x = self.rearrange_out(x, B=B)
+
+        return x
+
+    def decode(self, x, **kwargs):
+        # is_video = (x.ndim == 5)
+
+        B = x.shape[0]
+        # x: (B, Z, T, H, W)
+        # x = rearrange(x, "B Z T H W -> (B T) Z H W")
+        x = self.rearrange_in(x)
+
+        if self.micro_batch_size is None:
+            x = self.module.decode(x / self.scale_factor)
+        else:
+            # NOTE: cannot be used for training
+            bs = self.micro_batch_size
+            x_out = []
+            for i in range(0, x.shape[0], bs):
+                x_bs = x[i : i + bs]
+                x_bs = self.module.decode(x_bs / self.scale_factor)
+                x_out.append(x_bs)
+            x = ops.cat(x_out, axis=0)
+
+        # x = rearrange(x, "(B T) Z H W -> B Z T H W", B=B)
+        x = self.rearrange_out(x, B=B)
+
+        return x
+
+    def get_latent_size(self, input_size):
+        latent_size = []
+        for i in range(3):
+            # assert (
+            #     input_size[i] is None or input_size[i] % self.patch_size[i] == 0
+            # ), "Input size must be divisible by patch size"
+            latent_size.append(input_size[i] // self.patch_size[i] if input_size[i] is not None else None)
+        return latent_size
+
+
+class VideoAutoencoderPipelineConfig(PretrainedConfig):
```

Comment:
I don't think it's good to use `PretrainedConfig` with our project. It's better to move these parameters under `__init__` in `VideoAutoencoderPipeline`.